xla/pjrt/c/pjrt_c_api_gpu_test.cc

```diff
@@ -507,9 +507,13 @@ TEST(PJRTGpuDeviceTopologyTest, CreateExplicitGpuTopologyAndTargetConfig) {
       reinterpret_cast<const PJRT_TopologyDescription*>(args.topology);
   ASSERT_NE(pjrt_topology, nullptr);
 
+#ifdef TENSORFLOW_USE_ROCM
```

Comment:
Please do not use macros in new XLA code, we're actively trying to remove them.

Use a run-time call to determine the platform. Not sure if there is a better approach here, but in the worst case you can check that platform_id is either xla::RocmId() or xla::CudaId() and check that the platform name is consistent.

Reply:
Another comment: On line 492, this creates an explicit target_config that uses `platform_name: CUDA` (line 479). Is this supposed to work with ROCm? That doesn't seem right. If yes, then could you please write a comment right before line 490, to explain why this is OK?